### PR TITLE
4622 Update NR input/validation

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#4622

*Description of changes:*
- updated NR number v-text-field
- updated formatNrNumber()
- updated formatIncorporationNumber()
- updated validateNameRequestNumber()
- added/updated unit tests
- app version = 1.1.16

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/manage-business/AddNameRequestForm.vue [hunk 1]
```diff
@@ -4,32 +4,29 @@
       <fieldset>
         <legend hidden>Name Request Number and Applicant Phone Number or Email Address</legend>
         <v-text-field
-          filled
+          ref="nrNumber"
+          filled persistent-hint validate-on-blur
```
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AddNameRequestForm.vue:8]: `req` is mutually exclusive with validate-on-blur (and checked via a rule anyway).

`validate-on-blur` is used to perform rule validation when the user leaves this component.

File: auth-web/src/components/auth/manage-business/AddNameRequestForm.vue [hunk 2]
```diff
@@ -181,8 +182,10 @@ export default class AddNameRequestForm extends Vue {
     }
   }
 
-  private formatNrNumber (): void {
+  protected async formatNrNumber (): Promise<void> {
     this.nrNumber = CommonUtils.formatIncorporationNumber(this.nrNumber, true)
+    await Vue.nextTick() // wait for component to update
+    this.$refs.nrNumber.validate()
```
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AddNameRequestForm.vue:188]: When the user leaves the NR input component...
1. format the NR
2. wait for component to update
3. apply validation using rules

See ticket for test cases.

File: auth-web/src/util/common-util.ts [hunk 3]
```diff
@@ -13,9 +13,8 @@ export default class CommonUtils {
   }
 
   // formatting incorporation number according to the length of numbers
-  static formatIncorporationNumber (incorpNum:string, isNR: boolean = false, numLength?:number):string {
+  static formatIncorporationNumber (incorpNum: string, isNR = false, numLength = 7):string {
     if (!incorpNum) return null
-    numLength = numLength || 7 // optional: go with '7' if nothing specified
```
Comment (severinbeauvais) [inline auth-web/src/util/common-util.ts:18]: Now done in line 16.

**Review by lewischenstudio — APPROVED** [review]
LGTM

**Comment by seeker25** [comment]
Any dependencies on this one? I don't think so

**Review by seeker25 — APPROVED** [review]
LGTM

**Pull request merged**